src/domain/accounts/domain/src/repos/account_repository.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use internal_error::InternalError;
+use opendatafabric::{AccountID, AccountName};
+use thiserror::Error;
+
+use crate::Account;
+
+///////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait AccountRepository: Send + Sync {
+    async fn create_account(&self, account: &Account) -> Result<(), CreateAccountError>;
+
+    async fn get_account_by_id(
+        &self,
+        account_id: &AccountID,
+    ) -> Result<Account, GetAccountByIdError>;
+
+    async fn find_account_id_by_email(
+        &self,
+        email: &str,
+    ) -> Result<Option<AccountID>, FindAccountIdByEmailError>;
```

Comment:
With multiple providers active same email may map to multiple accounts.

Reply:
This will fail to register. We have a "unique" index constraint on email field.

Reply:
Shouldn't this result in briding, thus, reusing accounts from same person?